pkg/providers/vsphere/vsphere.go

```diff
@@ -431,6 +435,35 @@ func (p *vsphereProvider) SetupAndValidateCreateCluster(ctx context.Context, clu
 		logger.Info("Skipping check for whether control plane ip is in use")
 	}
 
+	var passed bool
+	var err error
+	vuc := config.NewVsphereUserConfig()
+
+	if passed, err = p.validator.validateUserPrivs(ctx, vSphereClusterSpec, vuc); err != nil {
+		return err
+	} else if passed {
+		s := fmt.Sprintf("%s user vSphere privileges validated", vuc.EksaVsphereUsername)
+		logger.MarkPass(s)
+	}
+
+	if len(vuc.EksaVsphereCPUsername) > 0 && vuc.EksaVsphereCPUsername != vuc.EksaVsphereUsername {
+		if passed, err = p.validator.validateCPUserPrivs(ctx, vSphereClusterSpec, vuc); err != nil {
+			return err
+		} else if passed {
+			s := fmt.Sprintf("%s user vSphere privileges validated", vuc.EksaVsphereCPUsername)
+			logger.MarkPass(s)
+		}
+	}
+
+	if len(vuc.EksaVsphereCSIUsername) > 0 && vuc.EksaVsphereCSIUsername != vuc.EksaVsphereUsername {
+		if passed, err = p.validator.validateCSIUserPrivs(ctx, vSphereClusterSpec, vuc); err != nil {
+			return err
+		} else if passed {
+			s := fmt.Sprintf("%s user vSphere privileges validated", vuc.EksaVsphereCSIUsername)
+			logger.MarkPass(s)
+		}
+	}
+
```

Comment:
This changed as well. We now only mark passed if the validation returns successfully.

This is where I would ideally like to place the `logger.MarkWarning` calls, but writing tests for this bit of code involves a lot of setup.